lib/clusterable/holding.rb

```diff
@@ -110,6 +110,18 @@ def update_key
       as_document.slice(*(fields.keys - EQUALITY_EXCLUDED_FIELDS)).hash
     end
 
+    def matches_commitment?
+      # check if there is a commitment on the same cluster
+      # with the same org & local_id
+      cluster.commitments.select do |spc|
```

Comment:
This is a little bit more questionable since here this introduces a dependency on the internal structure of `Commitment` that we didn't previously have, and reaches through the cluster to get them. You might consider something like:

```ruby

class Holding
  def matches_commitment?(commitment)
    commitment.organization == organization && commitment.local_id == local_id
  end
end

class Commitment
  def matches_holding?(holding)
    holding.organization == organization && holding.local_id = local_id
  end
end
```

and then defining a method on cluster that (essentially) joins them via merge-sort:
* sorts holdings & commitments by organization & local ID
* walks through the list of holdings & commitments to align them
* produce a combined list of holdings & commitments to operate on

(I haven't fully thought this through, obviously.)

This is a little bit of an awkward consequence of the mongodb model that would almost certainly be cleaner in a relational model.